### DISCUSSION TITLE
Maui on harmony os

### DISCUSSION THREAD

**Discussion by persiabudi28** [dotnet/maui#15536] [Q&A]
Maui can running on harmony os or huawei tab/ mobile phone?

**Comment by Xyncgas** [comment]
it's just android, so if they let you run MAUI you can.
Who's they you ask that's a really good question

**Reply by John0King** [reply]
@Xyncgas   not anymore ,  OpenHarmony 4.0 will completly remove  AOSP code

**Reply by Varorbc** [reply]
@Eilon Does the Maui team have a plan to support it?

**Reply by Eilon** [reply]
I don't think there's any plan for HarmonyOS at this time.

**Reply by tjiakwokyung28** [reply]
million people using harmonyos next in the future, this big market for application development and make money, dont focus on android, ios. tizen is dead

**Reply by Seanxwy** [reply]
This is a very good emerging market and Microsoft should not miss the HarmonyOS ecosystem. This should not be taken lightly, either for developers or for the market. Hope MAUI will take note and support this morning.

**Reply by i-py-cs** [reply]
> 百万人使用HarmonyOS的下一个未来，这个大市场对于应用开发和赚钱，不要把重点放在Android、iOS上。Tizen 死了

你说的没错但是安卓是以亿为单位的

**Reply by tjiakwokyung28** [reply]
In the future, there will be three major operating systems for smartphones: Android, Huawei(harmony next), and iOS.

**Reply by xiangxingze** [reply]
There is no point, now whether it is the application market, or the recruitment market are deliberately going to .net/C# language, developers are less and less, the enterprise does not accept .net/C#, MVP also did not play a leading role, and even take the MVP gimmick to sell classes to cut leeks!

**Reply by isapi** [reply]
> > 百万人使用HarmonyOS的下一个未来，这个大市场对于应用开发和赚钱，不要把重点放在Android、iOS上。Tizen 死了
> 
> 你说的没错但是安卓是以亿为单位的

别只盯着手机和平板，越来越多的国产开发板都支持鸿蒙了，后面是巨大的智能设备市场，可不止百万级。

**Reply by caidingding233** [reply]
> it's just android, so if they let you run MAUI you can.
> 
> Who's they you ask that's a really good question

If u think that it is just android, u should ask yourself, are u abandoned in the 2019?! Bro now 2025, 2026 is coming! Or are u Lei Jun's Mi Monkey Bot. If so, please tell me that how can I earn that money?

**Comment by tjiakwokyung28** [comment]
https://developer.huawei.com/consumer/en/doc/HMS-Plugin-Library/xamarin-sdk-download-0000001050768441

**Comment by SunshineSpring666** [comment]
I see the Harmony OS library for Xamarin, but now we need the MAUI support for Harmony OS 4+. Is Microsoft or Huawei plan to provide official library/support in recent months? Thanks.

**Reply by tjiakwokyung28** [reply]
if maui running on harmony os, maui really multi platform, 1st tehnology in the world for running any machine

**Comment by Seanxwy** [comment]
Hope to support HarmonyOS quickly. The user ecology of Hongmeng is a very potential ecology. Is a more potential ecosystem than Android, iOS. I can't wait to use MAUI+BLAZOR to develop applications.

**Comment by JohnYang1210** [comment]
Also hope to support HarmonyOS as early as possible!

**Comment by Xyncgas** [comment]
XZ backdoor

**Reply by Seanxwy** [reply]
What do you mean?

**Comment by tjiakwokyung28** [comment]
yes like xamarin, we hope maui to support HarmonyOS as early as possible!

**Comment by charleypeng** [comment]
yes, since harmoneyos next which supposed to be released in september will not support android apps, we really hope that Maui will adapt harmoneyos soon.

**Reply by tjiakwokyung28** [reply]
harmony os next using micro kernel, see https://en.m.wikipedia.org/wiki/Microkernel

**Comment by tjiakwokyung28** [comment]
China's productive population in 2024 will be 1.4 billion, this is a very large share of applications on huawei os, Maui must take part of this number. Microsoft should not repeat the same thing as in the case of Android, Nokia and Windows Phone

**Comment by Seanxwy** [comment]
This is a very good emerging market and Microsoft should not miss the HarmonyOS ecosystem. This should not be taken lightly, either for developers or for the market. Hope MAUI will take note and support this morning.

**Comment by charleypeng** [comment]
I noticed that flutter is now working on harmonyos version, hope that maui will catch up soon

**Comment by tjiakwokyung28** [comment]
Tizen OS is mostly used on smart TVs, other hardware is almost not used, even smartphones that use Tizen are also few, in this case the share of Tizen is very small and in the future it could die, while Huawei OS is used on smart phones, TVs, cars, electronic devices with large market share. Therefore, Net Maui should not repeat Microsoft's mistake with Windows Phone and Nokia which did not enter the Linux market, so that in the end Samsung took the market with Android, while Windows Phone and Nokia lost. Maui has to go to harmonynext OS so Maui can run on all machines and developers can enter any market to make money.

**Reply by Seanxwy** [reply]
Very much. I agree with you

**Comment by giuseppenovielli** [comment]
@davidortinau What do you think?

**Reply by Seanxwy** [reply]
@davidortinau  @Eilon What do you think?

**Reply by Varorbc** [reply]
@davidortinau @Eilon What do you think?

**Reply by charleypeng** [reply]
it is a good thing to see Maui on different platform

**Reply by Xyncgas** [reply]
Linux

**Comment by vinajava** [comment]
If you want to develop mobile apps using .NET MAUI, get an iOS or Android phone. Huawei has already decided that HarmonyOS will not support Android app, and instead intensifies development of HarmonyOS native app ecosystem. So I do not think Huawei will accept any .NET MAUI app in HarmonyOS. You do not have choice here.

**Comment by Seanxwy** [comment]
HarmonyOS is already the second largest operating system. In the Chinese market, it has surpassed the market share of ios. And the future is full of potential. Unity3d is already planning to support HarmonyOS, and MAUI should provide support sooner rather than later. To help developers achieve true cross-platform development.
I think the simplest pattern is based on MAUI+BLAZOR.

**Reply by sichenmy** [reply]
搞笑呢吧，我从农村到城市就没见过一个鸿蒙设备

**Comment by tjiakwokyung28** [comment]
In the future, there will be three major operating systems for smartphones: Android, Huawei, and iOS. in (1st position = android, 2nd = harmony next and 3rd = ios), or (1st = harmony next, 2nd = android, 3rd = ios).

**Reply by i-py-cs** [reply]
hello,Are you belong to .NET MAUI ？
How do you know the news?
It makes me happy a lot 
     from China

**Reply by John0King** [reply]
@tjiakwokyung28  is only talk about the OS  usage of future , not about MAUI I think. (IMO, it's not future, it is now)

**Comment by bxjg1987** [comment]
In China, HarmonyOS is very important and even receives first level support from the government. It has huge market potential. Should we consider supporting it.
At present, Unity3D has launched a special edition for China and supports HarmonyOS. We hope that Maui can evaluate its support for HarmonyOS.

**Comment by netty2019** [comment]
HarmonyOS has a very promising future, and I hope it will be supported sooner.

**Reply by Yu-Core** [reply]
So are people who use Apple, Samsung, Oppo, Vivo, Xiaomi low-end users? Stop talking useless nonsense. If you have the time, why not take a look at how the Unity TuanJie engine supports OpenHarmony? Maui needs your PR
https://developer.unity.cn/projects/66f3df8fedbc2a5b7602529d

**Reply by netty2019** [reply]
> So are people who use Apple, Samsung, Oppo, Vivo, Xiaomi low-end users? Stop talking useless nonsense. If you have the time, why not take a look at how the Unity TuanJie engine supports OpenHarmony? Maui needs your PR
> https://developer.unity.cn/projects/66f3df8fedbc2a5b7602529d
> 

Sorry, the comment has been revised. I hope we can work together to make MAUI develop even better.

**Reply by charleypeng** [reply]
actually,flutter now can run in harmoneyos next by the harmoneyos community

**Comment by Yannikk1996** [comment]
We also need to target the chinese market. Harmony support would be great.

**Comment by CeSun** [comment]
Seeing so many people paying attention to the support of MAUI in HamonyOS, I can't help but ask: 

What have you done?
Have you made any attempts to run .NET on HarmonyOS?

If you don't do anything yourself, why ask others to do it?

Especially for a friend with a certain ID, I have seen you asking others to support HamronyOS in many .net related projects. Why don't you start with what you can do? The meaning of the community is that everyone can contribute, not asking others to do something for you.

**Reply by ghkamfung** [reply]
Because not everyone is capable to develop MAUI framework/platform. Asking experts in certain area is normal for downstream user of a framework/platform.

Similarly, you can develop app on Android does not mean you can develop Android Kernel. 

I think support to HarmonyOS is better early than late.

**Reply by CeSun** [reply]
> Because not everyone is capable to develop MAUI framework/platform. Asking experts in certain area is normal for downstream user of a framework/platform.
> 
> Similarly, you can develop app on Android does not mean you can develop Android Kernel.
> 
> I think support to HarmonyOS is better early than late.

I understand that everyone wants to use their favorite language or framework on their favorite platform. But I think that endless inquiries and even some people questioning why maintainers don't support XXX are a drain on the enthusiasm of open source project maintainers.

Although not everyone has the ability to develop a framework, everyone can make the best efforts they can for what they love. Unfortunately, I don't see the efforts made by those who ask "When will HarmonyOS be supported" in various open source projects every day.

Even providing relevant documentation is one of the efforts

**Reply by tjiakwokyung28** [reply]
Microsoft must learn from Blackberry's past mistakes where it chose Blackberry OS instead of Android OS, which had a brighter and bigger market, in the end Blackberry lost to Samsung. Now the Tizen market is far from Harmony OS, in the future Harmony OS will be in control, no one knows, as developers, business people, we hope Maui can run on all OS like Maui's initial slogan,writing code one and running on multi platforms

**Reply by charleypeng** [reply]
what i do not agree is that we all know dotnet and maui is somehow dominated by microsoft and if they do not have any plan to support harmonyos(like what they did on linux),the community will hardly pay any attention on it. and yes we are just normal developers with pasions on technology, but this doesn't mean that we can not express our demonds and thoughts here. once the community started support project, we can try our best to help with it.

**Reply by CeSun** [reply]
> what i do not agree is that we all know dotnet and maui is somehow dominated by microsoft and if they do not have any plan to support harmonyos(like what they did on linux),the community will hardly pay any attention on it. and yes we are just normal developers with pasions on technology, but this doesn't mean that we can not express our demonds and thoughts here. once the community started support project, we can try our best to help with it.

Aren’t Flutter, which is led by Google, and QT, which is led by QT Company, both community-supported HarmonyOS?
As a member of the. NET community, why can't questioners be like Flutter and QT communities?

>  once the community started support project, we can try our best to help with it.

Don't wait for others to start, start doing what you can too

**Reply by charleypeng** [reply]
you'r funny,if i can start right now i will start and pull what i did. but i know i am not capable of it. howerver it does't mean i can not have passion about it. we are just giving suggestions to the dotnet community to support harmonyos as we love dotnet and csharp, not asking or commanding the dotnet community to do things. to be clear, i also wrote some suggestions to the flutter community and they replied in a more polite way without arrogance. p.s. yes flutter and qt is supported by the community in harmonyos,and it is working just fine, but where is the community supported maui on linux? i remember microsoft put this info in the maui official page but after all these years we see nothing, so we still need these tech giants to be envoled in the community supported projects.

**Reply by CeSun** [reply]
> you'r funny,if i can start right now i will start and pull what i did. but i know i am not capable of it. howerver it does't mean i can not have passion about it. we are just giving suggestions to the dotnet community to support harmonyos as we love dotnet and csharp, not asking or commanding the dotnet community to do things. to be clear, i also wrote some suggestions to the flutter community and they replied in a more polite way without arrogance. p.s. yes flutter and qt is supported by the community in harmonyos,and it is working just fine, but where is the community supported maui on linux? i remember microsoft put this info in the maui official page but after all these years we see nothing, so we still need these tech giants to be envoled in the community supported projects.

The community includes you and me, not Microsoft.

> yes flutter and qt is supported by the community in harmonyos,and it is working just fine, but where is the community supported maui on linux?

Who are you asking? Shouldn’t you ask those HarmonyOS and dotnet enthusiasts who keep asking in issues “Why doesn’t XX project support HarmonyOS”?

> but where is the community supported maui on linux? i remember microsoft put this info in the maui official page but after all these years we see nothing

One of the contributors of maui launched the maui project for the Linux platform a long time ago. Have you contributed any code?
https://github.com/jsuarezruiz/maui-linux


> howerver it does't mean i can not have passion about it.

So your passion is just to assign tasks to others on the grounds of incompetence?


**Allow me to ask again, the so-called nativeaot compiled exe can run on HarmonyOS, have you really tried it?**

Don't always exclude yourself from the community and then expect the community to serve you. You are a member of the community and you can start any attempt to accomplish this feat with HarmonyOS and dotnet enthusiasts. Microsoft has always maintained an open attitude, just like supporting LoogArch, provided that all the work cannot be pushed to Microsoft.

**Comment by vinajava** [comment]
Like I said, the main point here is whether Huawei accepts MAUI apps in HarmonyOS ? I do not think so, as Huawei has already strongly intensified development of HarmonyOS native app ecosystem. That means only HarmonyOS native apps are accepted. No choice.

**Reply by charleypeng** [reply]
but here is that flutter now can run in Harmony os and dotnet aot exe can also run. the question is how microsoft thought.would they pay attention to harmonyos, or they just let what is happening and doing nonthing until they were left behind.

**Comment by davidortinau** [comment]
I see several comparisons here to Tizen. If it's not already clear, the Tizen support in .NET MAUI is contributed by and maintained by Samsung. I'd welcome similar for HarmonyOS NEXT.

**Comment by tjiakwokyung28** [comment]
https://www.phonearena.com/news/android-apps-can-run-on-harmony-os-next_id165637

**Reply by tjiakwokyung28** [reply]
android running with emulator in harmonyOs 🤦‍♂️🤦‍♂️🤦‍♂️

**Comment by caidingding233** [comment]
Well here are someone made that. They used Avalonia to make the MAUI compatible with HarmonyOS.
